rasterio/windows.py

```diff
@@ -30,6 +42,15 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+def toranges(window):
```

Comment:
since this is a new function, it seems like it should be a method of the `Window` class instead, since no previously existing case will be using it.  Then we just need to make sure any new cases now using this have first normalized input to a `Window`.  Make sense?